modules/document_repository/ajax/documentEditUpload.php

```diff
@@ -147,4 +170,71 @@
     }
 }
 
+/**
+ * Handles the version inspection process.
+ * Generates correct proceeding version in some cases.
+ *
+ * @param string $version the version to use for uploading file.
+ * @param array  $items   the array of existing files (name & version).
+ *
+ * @return string $version.
+ */
+function versionInspectionGenerator($version, $items)
```

Comment:
I don't know if I fully agree with the logic here. It seems like you'd be forcing `\d.\d.(\*|\d)`, which could also incorrectly default to `0.0.1` if there are already files in the user_uploads folder.

I feel like generated IDs are a way better way of handling unique file identifiers rather than version #s (however I get that this isn't ideal for file directories). I also don't think we should force version #s on user files, because they may get confused when a random 0.0.1 shows up.

Reply:
I feel it should be up to the user to enter the version of a file that they uploaded. If I want to upload a file called hello_toto.pdf and append the version 1.0 to it, it will not consider what I input in the upload file form and number it 0.0.1 which is not what I wanted and is confusing. I think you should use a different field than the version field of the document repository to version the uploaded file itself in case of duplication of files.

Reply:
@cmadjar That definitely makes more sense!